tests/rptest/services/redpanda_connect.py

```diff
@@ -0,0 +1,132 @@
+    """
+    Redpanda Connect service managed by RPK
+    """
+    PERSISTENT_ROOT = "/var/lib/redpanda_connect/"
+    RPK_BIN = "rpk"
+    LOG_FILE = os.path.join(PERSISTENT_ROOT, "connect.log")
+    logs = {
+        "redpanda_connect_log": {
+            "path": LOG_FILE,
+            "collect_default": True
+        }
+    }
+
+    redpanda: RedpandaService
+    _pid: typing.Optional[int]
+    logging_config = """
+logger:
+    level: DEBUG
+    format: logfmt
+    add_timestamp: true
+    static_fields:
+        '@service': redpanda-connect
+
+"""
+
+    def __init__(self, context: TestContext, redpanda: RedpandaService,
+                 configuration: str):
+        super().__init__(context, num_nodes=1)
+        self.redpanda = redpanda
+
+        self._pid = None
+        self._cfg = configuration
+
+    def _rpk_binary(self):
+        # NOTE: since this runs on separate nodes from the service, the binary
+        # path used by each node may differ from that returned by
+        # redpanda.find_binary(), e.g. if using a RedpandaInstaller.
+        rp_install_path_root = self.context.globals.get(
+            "rp_install_path_root", None)
+        return f"{rp_install_path_root}/bin/rpk"
+
+    def clean_node(self, node):
+        self.logger.debug(f"{self.__class__.__name__}.clean_node")
+        node.account.kill_process(self.RPK_BIN, clean_shutdown=False)
+
+        if node.account.exists(self.PERSISTENT_ROOT):
+            node.account.remove(self.PERSISTENT_ROOT)
+        self.logger.info("Uninstalling redpanda-connect from %s",
+                         node.account.hostname)
+
+        self._execute_cmd(node, ["uninstall"])
+
+    def _execute_cmd(self, node: ClusterNode, cmd: list):
+        return node.account.ssh_output(self._connect_command(cmd))
+
+    def _connect_command(self, cmd: list):
+        return " ".join([self._rpk_binary(), "connect"] + cmd)
+
+    def start_node(self, node):
+        node.account.mkdirs(self.PERSISTENT_ROOT)
+        self.logger.info("Installing redpanda-connect on %s",
+                         node.account.hostname)
+        self._execute_cmd(node, ["install"])
+        cfg_file = os.path.join(self.PERSISTENT_ROOT, "connect.yaml")
+        effective_cfg = self._cfg + self.logging_config
+        self.logger.debug(
+            f"Writing Redpanda Connect configuration: {effective_cfg}")
+
+        node.account.create_file(
+            os.path.join(self.PERSISTENT_ROOT, "connect.yaml"), effective_cfg)
+
+        node.account.ssh(
+            f"{self._connect_command(['run', cfg_file])}  1>> {self.LOG_FILE} 2>> {self.LOG_FILE} &"
+        )
+
+        def _ready():
+            r = requests.get(f"http://{node.account.hostname}:4195/ready")
+            return r.status_code == 200
+
+        wait_until(_ready,
+                   timeout_sec=30,
+                   backoff_sec=0.5,
+                   err_msg="Redpanda Connect failed to start",
+                   retry_on_exc=True)
```

Comment:
This is technically racy. Connect could finish while in a backoff interval.

Better would be to to use Streams mode: https://docs.redpanda.com/redpanda-connect/guides/streams_mode/using_rest_api/


Basically start rpcn in streams mode then wait for ready to return true.

After that post the config and wait node can looks for `active` being false meaning the pipeline finished